window/src/os/wayland/window.rs

```diff
@@ -547,12 +549,23 @@ impl WaylandWindowInner {
             }
         }
 
-        if let Some((w, h)) = pending.configure.take() {
+        if let Some((mut w, mut h)) = pending.configure.take() {
             if self.window.is_some() {
                 let factor = get_surface_scale_factor(&self.surface);
 
-                let pixel_width = self.surface_to_pixels(w.try_into().unwrap());
-                let pixel_height = self.surface_to_pixels(h.try_into().unwrap());
+                let mut pixel_width = self.surface_to_pixels(w.try_into().unwrap());
+                let mut pixel_height = self.surface_to_pixels(h.try_into().unwrap());
+
+                if self.window_state.can_resize() {
+                    if let Some((x, y)) = self.resize_increments {
+                        let desired_pixel_width = pixel_width - (pixel_width % x as i32);
+                        let desired_pixel_height = pixel_height - (pixel_height % y as i32);
+                        w = self.pixels_to_surface(desired_pixel_width) as u32;
```

Comment:
I've been semi-idly glancing at this with the TV on in the background, so these may not be fully formed thoughts...

This transform to and from the scaled size feels tricky to me, especially with it leading to the use of glScissor.

I wonder if it would be a bit simpler overall if we could just set `pixel_width = desired_pixel_width` instead of this adjustment here.

To make that work, I think `resize_increments` needs to be recorded in terms of the scaled size, and we could perform that adjustment in `set_resize_increments`.

That way we could avoid the `crop_area` fanning out into `Dimensions` and `glScissor`.

What do you think?

Reply:
This final code is definitely not the first thing I've tried :)

- Rounding the logical size up to "unscaled" cell size (divided by the factor) leads to *huge* gaps, because the rounding error accumulates over *all* the rows/columns
- Using the desired size as pixel (buffer) size and just "also" rounding the logical size leads to a mismatch that results in nothing getting rendered (stuck on the first frame, forever quarter size with one square cursor)
  - That was tested with the bad rounding that results in the huge gaps, but a mismatch's a mismatch, I'm almost certain that a 1px mismatch would result in the same issue
- There's a very simple option of just rounding the logical size up to the physical cell size… resulting in steps of 2 rows/cols on 2x scale, etc. which is not great